### PR TITLE
fix: Upstream information is lost when editing services

### PR DESCRIPTION
**Pull request by guoqqqi**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [x] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues

___
### Bugfix
- Description
1. Create the service and select the existing upstream
image

2. Edit, upstream information is lost
image

- How to fix?
Added `shouldUpdate` method to update form information.
___

### CODE REVIEW

**Comment by liuxiran** [comment]
it worked well when I ran this pr locally :100: 

how about add to a test case to cover this situation?

**Comment by juzhiyuan** [comment]
waiting for testcase

File: web/cypress/integration/service/edit-service-with-upstream.spec.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* eslint-disable no-undef */
+
+context('edit service with upstream', () => {
```
Comment (juzhiyuan) [inline web/cypress/integration/service/edit-service-with-upstream.spec.js:19]: `Edit Service with Upstream`

File: web/cypress/integration/service/edit-service-with-upstream.spec.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    create_upstream_success: 'Create Upstream Successfully',
+    delete_service_success: 'Delete Service Successfully',
+    delete_upstream_success: 'Delete Upstream Successfully',
+  };
+
+  beforeEach(() => {
+    cy.login();
+  });
+
+  it('should create test upstream', () => {
```
Comment (juzhiyuan) [inline web/cypress/integration/service/edit-service-with-upstream.spec.js:45]:
```suggestion
  it('should create a test upstream', () => {
```

File: web/cypress/integration/service/edit-service-with-upstream.spec.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+    cy.get(domSelector.name).type(data.test_upstream);
+    cy.get(domSelector.nodes_0_host).type(data.ip);
+    cy.contains('Next').click();
+    cy.contains('Submit').click();
+    cy.get(domSelector.notification).should('contain', data.create_upstream_success);
+    cy.url().should('contains', 'upstream/list');
+  });
+
+  it('should create test service', () => {
```
Comment (juzhiyuan) [inline web/cypress/integration/service/edit-service-with-upstream.spec.js:58]: `a test xx`

File: web/cypress/integration/service/edit-service-with-upstream.spec.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    cy.visit('/');
+    cy.contains('Service').click();
+    cy.contains('Create').click();
+    cy.get(domSelector.name).type(data.service_name);
+    cy.get(domSelector.desc).type(data.desc);
+    cy.get(domSelector.custom_upstream).click();
```
Comment (juzhiyuan) [inline web/cypress/integration/service/edit-service-with-upstream.spec.js:64]: `custom_upstream` -> `upstream_selector`?

File: web/cypress/integration/service/edit-service-with-upstream.spec.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    cy.contains(data.test_upstream).click();
+    cy.get(domSelector.input).should('be.disabled');
+
+    cy.contains('Next').click();
+    cy.contains('Next').click();
+    cy.contains('Submit').click();
+    cy.get(domSelector.notification).should('contain', data.create_service_success);
+  });
+
+  it('should edit the service', () => {
```
Comment (juzhiyuan) [inline web/cypress/integration/service/edit-service-with-upstream.spec.js:74]: doesn't edit the service actually..

File: web/cypress/integration/service/edit-service-with-upstream.spec.js [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+    cy.get(domSelector.nodes_0_host).click().should('value', data.ip);
+    cy.get(domSelector.input).should('be.disabled');
+
+    cy.get(domSelector.test_upstream).click();
+    cy.contains('Custom').click();
+    cy.get(domSelector.nodes_0_host).should('not.be.disabled');
+  });
+
+  it('should delete service and upstream', () => {
```
Comment (juzhiyuan) [inline web/cypress/integration/service/edit-service-with-upstream.spec.js:90]: both and

**Comment by LiteSun** [comment]
This change also affects the upstream part of the consumer and route, so keep an eye on that. @guoqqqi

**Comment by guoqqqi** [comment]
> This change also affects the upstream part of the consumer and route, so keep an eye on that. @guoqqqi

Thanks for the suggestion, I've tested them locally and they all work fine.

File: web/cypress/integration/route/online-debug.spec.js [hunk 7]
```diff
@@ -61,6 +61,7 @@ context('Online debug', () => {
       cy.get(domSelector.uriInput).type(uri);
       cy.contains(routeLocaleUS['page.route.button.send']).click({force: true});
 
+
```
Comment (LiteSun) [inline web/cypress/integration/route/online-debug.spec.js:64]: remove extra line.

**Review by juzhiyuan — APPROVED** [review]

**Review by LiteSun — APPROVED** [review]

**Pull request merged**